tests/unit/dag/ops/test_rename.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+import pandas as pd
+import pytest
+
+from merlin.core.compat import cudf
+from merlin.dag import ColumnSelector
+from merlin.dag.ops.rename import Rename
+from merlin.table import TensorTable
+
+transformables = [pd.DataFrame, TensorTable]
+if cudf:
+    transformables.append(cudf.DataFrame)
+
+
+@pytest.mark.parametrize("transformable", transformables)
+def test_rename(transformable):
+    df = transformable({"x": np.array([1, 2, 3, 4, 5]), "y": np.array([6, 7, 8, 9, 10])})
+
+    selector = ColumnSelector(["x", "y"])
+
+    op = Rename(f=lambda name: name.upper())
+    transformed = op.transform(selector, df)
+    expected = transformable({"X": np.array([1, 2, 3, 4, 5]), "Y": np.array([6, 7, 8, 9, 10])})
+    assert all(transformed == expected)
+
+    op = Rename(postfix="_lower")
+    transformed = op.transform(selector, df)
+    expected = transformable(
+        {"x_lower": np.array([1, 2, 3, 4, 5]), "y_lower": np.array([6, 7, 8, 9, 10])}
+    )
+    assert all(transformed == expected)
+
+    selector = ColumnSelector(["x"])
+
+    op = Rename(name="z")
+    transformed = op.transform(selector, df)
+    expected = transformable({"z": np.array([1, 2, 3, 4, 5])})
+    assert all(transformed == expected)
+
+    op = Rename(f=lambda name: name.upper())
+    transformed = op.transform(selector, df)
+    expected = transformable({"X": np.array([1, 2, 3, 4, 5])})
+    assert all(transformed == expected)
```

Comment:
This is will return True for DataFrames as long as the columns are different because the return type of `DataFrame == DataFrame` is a DataFrame, and `all(DataFrame)` iterates through the column names and asserts they are all truthy values. [DataFrame.eq](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.DataFrame.eq.html) performs element-wise equality.

If we mostly need to know if a DataFrame contains the same elements and don't need to operate on the element-wise equality result, maybe could use [DataFrame.equals](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.DataFrame.equals.html) or [pd.testing.assert_frame_equal](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.testing.assert_frame_equal.html#pandas.testing.assert_frame_equal) to assert dataframe equality instead.

If we implemented an `equals` method on TensorTable instead of element-wise `__eq__` we might not immediately need the `__eq__` for this use-case. However, if we do add that to `TensorTable` and would like to be consistent with the Series/DataFrame `== / __eq__` equivalent perhaps we would return a TensorTable from this with element-wise comparison At the moment it looks like we're returning a tensor type. 

If the motivation is about being able to write unit tests that transform both TensorTable and DataFrame inputs. Having a helper funciton in core could be another option `assert_frame_or_table_equal`.

Reply:
One benefit of `assert_frame_equal` over `DataFrame.equals` I can think of are error messages that explain why the dataframe doesn't match. `equals` returns True or False always and doesn't intend to raise exceptions or print messages, while `assert_frame_equal` is designed for testing and will explain why the dataframe doesn't match.


```python
pd.testing.assert_frame_equal(pd.DataFrame({"a": [1.0, 2.0]}), pd.DataFrame({"a": [1, 2]}))
# =>
# AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="a") are different
# 
# Attribute "dtype" are different
# [left]:  float64
# [right]: int64


pd.DataFrame({"a": [1.0, 2.0]}).equals(pd.DataFrame({"a": [1, 2]}))
# => False
```

Another feature is by default it will use an tolerance check to avoid small floating point differences

Reply:
This PR has been updated to add an `assert_transformable_equal` testing helper and to remove changes to `TensorTable` and `TensorColumn`'s equality definitions. Those will still need some changes to standardize them with DataFrames and Series, but those changes aren't required to get `Rename` moved into Core.